### DISCUSSION TITLE
Controlling tesselation id for each subscale tesselation

### DISCUSSION THREAD

**Discussion by danishTUE** [neperfepx/neper#920] [Q&A] [answered]
Hi Romain,

I am interested in creating a multiscale tesselation wherein I want to control the id (and thus the seed used for its generation) at subscale. As an example I am trying to get same arrangement of grains at subscale using a msfile. The script runs but I end up getting different spatial arrangement of sub-grains for each primary grain. Is it something that I can be achieved with Neper? For e.g. getting the same spatial arrangement of sub-grains for each primary grain? Here is the script of what I am using to try to achieve this:

```
neper -T -n from_morpho::2 -id "1::msfile(id_msfile_8.txt)" -dim 3 -domain 'cube(1,1,1)' -morpho "cube(2)::diameq:0.5,1-sphericity:lognormal(0.1,0.03)" -o multi_scale
neper -V multi_scale.tess -datacellcol "scaleid(1)" -datacelltrs 0.5 -print multi_scale
``` 
Many thanks in advance.

Regards,

Danish
[id_msfile_8.txt](https://github.com/user-attachments/files/17330100/id_msfile_8.txt)

**Comment by rquey** [comment, marked as answer]
The seed distribution does not depend only on the id at the scale of interest, but also of the id at the previous scales. So, the only solution I see is to load custom seed coordinates (and weights) for all subcells.  The seed coordinates (and weights) would be the same, except for the translation.
